lib/polymorphic_integer_type/extensions.rb

```diff
@@ -7,10 +7,22 @@ def belongs_to(name, scope = nil, options = {})
         options = scope if scope.kind_of? Hash
         integer_type = options.delete :integer_type
         super
-        if options[:polymorphic] && integer_type
-          mapping = PolymorphicIntegerType::Mapping[name]
+        if options[:polymorphic] && (integer_type || options[:polymorphic].is_a?(Hash))
+          mapping =
+            case integer_type
+            when true then PolymorphicIntegerType::Mapping[name]
+            when nil then options[:polymorphic]
+            else
+              integer_type
```

Comment:
when would we have this default?

Reply:
Good catch. Originally, the mapping was passed to `integer_type:`. We can drop this case.